crates/cxx-qt-lib/src/core/qurl.rs

```diff
@@ -6,16 +6,159 @@ use cxx::{type_id, ExternType};
+        #[rust_name = "error_string"]
+        fn errorString(self: &QUrl) -> QString;
+
+        /// Returns true if this URL contains a fragment (i.e., if # was seen on it).
+        #[rust_name = "has_fragment"]
+        fn hasFragment(self: &QUrl) -> bool;
+
+        /// Returns true if this URL contains a Query (i.e., if ? was seen on it).
+        #[rust_name = "has_query"]
+        fn hasQuery(self: &QUrl) -> bool;
+
+        /// Returns true if the URL has no data; otherwise returns false.
+        #[rust_name = "is_empty"]
+        fn isEmpty(self: &QUrl) -> bool;
+
+        /// Returns true if this URL is pointing to a local file path. A URL is a local file path if the scheme is "file".
+        #[rust_name = "is_local_file"]
+        fn isLocalFile(self: &QUrl) -> bool;
+
+        /// Returns true if this URL is a parent of child_url.
+        /// childUrl is a child of this URL if the two URLs share the same scheme and authority,
+        /// and this URL's path is a parent of the path of child_url.
+        #[rust_name = "is_parent_of"]
+        fn isParentOf(self: &QUrl, child_url: &QUrl) -> bool;
+
+        /// Returns true if the URL is relative; otherwise returns false.
+        /// A URL is relative reference if its scheme is undefined;
+        /// this function is therefore equivalent to calling scheme().is_empty().
+        #[rust_name = "is_relative"]
+        fn isRelative(self: &QUrl) -> bool;
+
+        /// Returns true if the URL is non-empty and valid; otherwise returns false.
+        #[rust_name = "is_valid"]
+        fn isValid(self: &QUrl) -> bool;
+
+        /// Returns the port of the URL, or defaultPort if the port is unspecified.
+        #[rust_name = "port_or"]
+        fn port(self: &QUrl, port: i32) -> i32;
+
+        /// Returns the result of the merge of this URL with relative. This URL is used as a base to convert relative to an absolute URL.
+        fn resolved(self: &QUrl, relative: &QUrl) -> QUrl;
+
+        /// Returns the scheme of the URL. If an empty string is returned,
+        /// this means the scheme is undefined and the URL is then relative.
+        ///
+        /// The scheme can only contain US-ASCII letters or digits,
+        /// which means it cannot contain any character that would otherwise require encoding
+        /// Additionally, schemes are always returned in lowercase form.
+        fn scheme(self: &QUrl) -> QString;
+
+        /// Sets the port of the URL to port.
+        ///
+        /// port must be between 0 and 65535 inclusive. Setting the port to -1 indicates that the port is unspecified.
+        #[rust_name = "set_port"]
+        fn setPort(self: &mut QUrl, port: i32);
+
+        /// Returns the path of this URL formatted as a local file path.
+        /// The path returned will use forward slashes, even if it was originally created from one with backslashes.
+        #[rust_name = "to_local_file"]
+        fn toLocalFile(self: &QUrl) -> QString;
```

Comment:
Could you rename this to `to_local_file_or_default`, then in the actual `to_local_file` implementation, check that it actually is a local file with `is_local_file` and use `Option<Self>` as the result value?
I think that would be the rust-y way to approach this, like with `value` on QVariant.

Similar to what you have done for `fragment`, etc.

Reply:
done